lib/web/websuite.go

```diff
@@ -0,0 +1,750 @@
+package web
```

Comment:
Unsure what others will think - but I'm quite uncomfortable with test helpers making their way into the packages themselves.  The lines quickly become blurred as to whats for the test and whats actually making this code tick in production (e.g something like `AddCSRFCookieToReq`). Could we do something like `lib/web/test` instead ?

Reply:
I am not all that comfortable with it myself. I did try to split it into a separate package, but circular dependencies on creating the `web.Handler` (from the `web.Config`) made that quite difficult (package web would need to use this new `webfixture` package, and it would need to use `web` to stand up the fixture). It might be possible to abstract that through an interface, but the `Config` struct is pretty unwieldy in its breadth.

I only went this route when I noted that `lib/auth/helper.go` contained `TestAuthServer` and similar.

Reply:
And sorry, I meant to note that in the PR description.

Reply:
One (imperfect) solution is to hide the testing utilities behind a build tag, which would only be used for testing. We may also have luck with the `internal` directory (https://go.dev/doc/go1.4#internalpackages), but I'm not sure we can use it in our case.

I also wonder if we may simply copy the code, either manually or with `go:generate` magic.